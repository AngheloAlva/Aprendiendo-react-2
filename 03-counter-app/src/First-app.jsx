/* eslint-disable react/prop-types */
import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle }) => {
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
    </>
  )
}

FirstApp.prototypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

FirstApp.defaultProps = {
  // title: 'Titulo por defecto',
  subtitle: 'Subtitulo'
}
