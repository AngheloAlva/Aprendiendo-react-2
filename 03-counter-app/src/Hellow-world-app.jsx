export function HellowWorldApp () {
  return (
    <h1>Hello world!!!</h1>
  )
}
