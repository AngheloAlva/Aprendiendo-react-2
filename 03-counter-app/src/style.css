body {
  background-color: #21232A;
  color: #fff;
  margin: 0;
  padding: 4rem 0;
  font-family: 'Roboto', sans-serif;
  width: 100vw;
  display: flex;
  align-items: center;
  justify-content: center;
}

button {
  background-color: #21232A;
  color: #fff;
  border: 1px solid #fff;
  padding: 10px 20px;
  border-radius: 5px;
  cursor: pointer;
  font-family: 'Roboto', sans-serif;
}

button:hover {
  background-color: #fff;
  color: #21232A;
}
