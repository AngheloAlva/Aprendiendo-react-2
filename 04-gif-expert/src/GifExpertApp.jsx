import { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = () => {
  // 'Batman', 'Villains',
  const [categories, setCategories] = useState(['Joker'])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return

    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h2>GifExpertApp</h2>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  )
}

export default GifExpertApp
