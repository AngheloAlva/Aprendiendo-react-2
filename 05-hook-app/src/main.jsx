import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import {
  createBrowserRouter,
  RouterProvider
} from 'react-router-dom'

import MainApp from './09-useContext/MainApp'
import AboutPage from './09-useContext/AboutPage'
import HomePage from './09-useContext/HomePage'
import LoginPage from './09-useContext/LoginPage'

const router = createBrowserRouter([
  {
    path: '/',
    element: <MainApp />,
    children: [
      { path: '/', element: <HomePage /> },
      { path: '/about', element: <AboutPage /> },
      { path: '/login', element: <LoginPage /> }
    ]
  }
])

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>
)
