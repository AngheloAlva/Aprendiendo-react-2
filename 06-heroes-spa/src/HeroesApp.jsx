const HeroesApp = () => {
  return (
    <h1 className='text-2xl font-bold'>HeroesApp</h1>
  )
}

export default HeroesApp
