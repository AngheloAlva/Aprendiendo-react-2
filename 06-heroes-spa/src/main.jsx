import React from 'react'
import ReactDOM from 'react-dom/client'
import { createBrowserRouter, RouterProvider } from 'react-router-dom'
// import HeroesApp from './HeroesApp'
import DcPage from './heroes/pages/DcPage'
import LoginPage from './auth/pages/LoginPage'
import Navbar from './heroes/components/Navbar'
import MarvelPage from './heroes/pages/MarvelPage'

import './index.css'
import SearchPage from './heroes/pages/SearchPage'
import HeroPage from './heroes/pages/HeroPage'

const router = createBrowserRouter([
  {
    path: '/',
    element: <Navbar />,
    children: [
      {
        path: '/marvel',
        element: <MarvelPage />
      },
      {
        path: '/dc',
        element: <DcPage />
      },
      {
        path: '/search',
        element: <SearchPage />
      },
      {
        path: '/hero/:heroId',
        element: <HeroPage />
      }
    ]
  },
  {
    path: '/login',
    element: <LoginPage />
  }
])

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>
)
