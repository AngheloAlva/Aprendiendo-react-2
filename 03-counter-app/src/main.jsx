import React from 'react'
import ReactDOM from 'react-dom/client'
// import { HellowWorldApp } from './hellow-world-app'
// import { FirstApp } from './first-app'
import './style.css'
import CounterApp from './Counter-app'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HellowWorldApp /> */}
    {/* <FirstApp title='Titulo epico' /> */}
    <CounterApp value={100} />
  </React.StrictMode>
)
