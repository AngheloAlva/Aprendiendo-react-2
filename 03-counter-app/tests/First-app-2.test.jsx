import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/first-app'

/* eslint-disable no-undef */
describe('Pruebas en First-app', () => {
  const title = 'Titulo epico'
  const subtitle = 'Subtitulo'

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()
  })

  test('Debe mostrar el titulo con el mensaje: "Hola, soy Goku"', () => {
    render(<FirstApp title={title} />)
    expect(screen.getByText(title)).toBeTruthy()
  })

  test('Debe mostrar el titulo en un <h1>', () => {
    render(<FirstApp title={title} />)
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toBe(title)
  })

  test('Debe mostrar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} subtitle={subtitle} />)
    expect(screen.getAllByText(subtitle).length).toBe(2)
  })
})
