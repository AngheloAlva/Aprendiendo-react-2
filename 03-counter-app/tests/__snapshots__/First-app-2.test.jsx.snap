// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en First-app Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titulo epico
  </h1>
  <p>
    Subtitulo
  </p>
  <p>
    Subtitulo
  </p>
</div>
`;
